### PR TITLE
Implement FlxFlicker.pause() and FlxFlicker.resume()

### PR DESCRIPTION
**Pull request by EliteMasterEric**
During the development of Funkin', I found the need to pause an active FlxFlicker without actively destroying it and recreating it. This PR adds that functionality.

### CODE REVIEW

**Comment by Geokureli** [comment]
Down to merge, but note that `FlxTween.flicker` now exists which has all the same tween methods for pausing, resuming and cancelling. It also allows for custom "ease" logic, so you can use one tween to do a flicker that changes frequency rather than using two calls, like you mentioned

**Comment by EliteMasterEric** [comment]
> Down to merge, but note that `FlxTween.flicker` now exists which has all the same tween methods for pausing, resuming and cancelling. It also allows for custom "ease" logic, so you can use one tween to do a flicker that changes frequency rather than using two calls, like you mentioned

Good to know, thanks for that.

File: flixel/effects/FlxFlicker.hx [hunk 1]
```diff
@@ -144,6 +144,22 @@ class FlxFlicker implements IFlxDestroyable
 		timer = new FlxTimer().start(interval, flickerProgress, Std.int(duration / interval));
 	}
 
+	/**
+	 * Temporarily pause the flickering, so it can be resumed later.
+	 */
+	public function pause():Void {
+		if (timer == null) return;
```
Comment (Geokureli) [inline flixel/effects/FlxFlicker.hx:151]:
```suggestion
		if (timer == null)
			return;
```

File: flixel/effects/FlxFlicker.hx [hunk 2]
```diff
@@ -144,6 +144,22 @@ class FlxFlicker implements IFlxDestroyable
...
+		timer.active = false;
+	}
+
+	/**
+	 * Resume the flickering after it has been temporarily paused.
+	 */
+	public function resume():Void {
+		if (timer == null) return;
```
Comment (Geokureli) [inline flixel/effects/FlxFlicker.hx:159]:
```suggestion
		if (timer == null)
			return;
		
```

**Review by Geokureli — CHANGES_REQUESTED** [review]
see suggestions

File: flixel/effects/FlxFlicker.hx [hunk 3]
```diff
@@ -144,6 +144,22 @@ class FlxFlicker implements IFlxDestroyable
 		timer = new FlxTimer().start(interval, flickerProgress, Std.int(duration / interval));
 	}
 
+	/**
+	 * Temporarily pause the flickering, so it can be resumed later.
+	 */
+	public function pause():Void {
```
Comment (Geokureli) [inline flixel/effects/FlxFlicker.hx:150]:
```suggestion
	public function pause():Void
	{
```

File: flixel/effects/FlxFlicker.hx [hunk 4]
```diff
@@ -144,6 +144,22 @@ class FlxFlicker implements IFlxDestroyable
...
+		if (timer == null) return;
+		timer.active = false;
+	}
+
+	/**
+	 * Resume the flickering after it has been temporarily paused.
+	 */
+	public function resume():Void {
```
Comment (Geokureli) [inline flixel/effects/FlxFlicker.hx:158]:
```suggestion
	public function resume():Void
	{
```

**Review by Geokureli — APPROVED** [review]

**Comment by Geokureli** [comment]
Thanks!

**Pull request merged**